src/ImageSharp/Formats/Png/PngDecoderCore.cs

```diff
@@ -1155,7 +1150,7 @@ private void ValidateChunk(in PngChunk chunk)
             this.crc.Update(chunk.Data.GetSpan());
 
             uint crc = this.ReadChunkCrc();
-            if (this.crc.Value != crc)
+            if (this.crc.Value != crc && chunk.IsCritical)
```

Comment:
what i dont like about this change: we calculate the crc for none critical chunks, but then ignore it.

Maybe we could always read the crc bytes on the start of the method instead?

Reply:
Yeah much better

Reply:
Yeah much better